OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs

```diff
@@ -109,7 +109,7 @@ public GameInfoStatsLogic(Widget widget, World world, OrderManager orderManager,
 
 					var flag = item.Get<ImageWidget>("FACTIONFLAG");
 					flag.GetImageCollection = () => "flags";
-					if (player == null || player.Stances[pp] == Stance.Ally || player.WinState != WinState.Undefined)
+					if (world.LocalPlayer == null || player.Stances[pp] == Stance.Ally || world.LocalPlayer.WinState != WinState.Undefined)
```

Comment:
IMO it would be better to change `player` to always be `world.LocalPlayer` instead. Having the score screen display tied to the spectator's selected visibility was an intentional but poorly considered feature.

The same arguments leading to changing this for the flags also applies to the mission objectives above.